src/coreclr/jit/hwintrinsic.h

```diff
@@ -70,175 +70,176 @@ enum HWIntrinsicCategory : uint8_t
 #else
 #error Unsupported platform
 #endif
+
 enum HWIntrinsicFlag : unsigned int
 {
     HW_Flag_NoFlag = 0,
 
     // Commutative
     // - if a binary-op intrinsic is commutative (e.g., Add, Multiply), its op1 can be contained
-    HW_Flag_Commutative = 0x1,
+    HW_Flag_Commutative = (1 << 0),
```

Comment:
This enum is muddled - it has common entries, then target specific, then more common entries (eg HW_Flag_SpecialSideEffect_Barrier).

I've reordered it, and then used defines for the target specific entries to make it easier to add common entries later. More common entires can be added after HW_Flag_CanBenefitFromConstantProp and only HW_Flag_PlatformBase needs updating

Reply:
If it is just cleanup, I would suggest having it as part of a separate PR.

Reply:
I did have a new flag and it caused the enum to run out of space, so I refactored. But then I removed the flag.

I'll revert these changes and put in a new PR.